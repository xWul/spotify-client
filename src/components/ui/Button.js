import React from 'react'
import styled from 'styled-components'

const Content = styled.div`
  width: 100%;
  padding: 5px;
`

const ButtonStyled = styled.button`
  cursor: pointer;
  background-color: #6290C8;
  color: #FFF;
  padding: 6px;
  width: 100%;
  border: 1px solid #CCC;
  border-radius: 3px;
  font-size: 16px;
  font-weight: bold; 
  text-align: center;
  text-transform: uppercase;
  &:hover {
    background-color: #376996;
  }
`

const Button = (props) => {
  return (
    <Content>
      <ButtonStyled {...props}/>
    </Content>
  )
}

export default Button
