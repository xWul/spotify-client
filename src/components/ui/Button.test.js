import React from 'react'
import renderer from 'react-test-renderer'
import Button from './Button'

test('Button renderer without crash', () => {
  const component = renderer.create(
    <Button type={'button'}>Testing</Button>,
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
