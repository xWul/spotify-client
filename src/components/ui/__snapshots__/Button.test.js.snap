// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renderer without crash 1`] = `
<div
  className="sc-bdVaJa LDKxg"
>
  <button
    className="sc-bwzfXH jAinUC"
    type="button"
  >
    Testing
  </button>
</div>
`;
